### DISCUSSION TITLE
Source code fetching when debug mainnet transaction

### DISCUSSION THREAD

**Discussion by Troublor** [remix-run/remix#1838] [Q&A] [answered]
I see remix is able to debug mainnet transaction if I use MetaMask's injected provider on mainnet. 

However, when I try with some transactions, like `0x9408a850428af7680922397e8be6aba2d4d371c5d9e2f532ca7833c93f6e5149`, there is a warning
```
Contract 0xdac17f958d2ee523a2206206994597c13d831ec7 not found in source code repository continuing without source code debugging.
```

But if I manually create a source code file for the contract and compile it, remix can find the source code when debugging. 

I wonder if there is a way to let remix automatically fetch contract verified source code from etherscan? 
Manually downloading is tedious and not realistic if the transaction involves several very complicated contracts.

**Comment by shamsup** [comment, marked as answer]
Is this meant for the Remix IDE? https://github.com/ethereum/remix-project

This repository is for the Remix JavaScript web framework, https://remix.run

**Reply by Troublor** [reply]
Oh right, sorry for the trouble caused. I got confused.
